storage/innobase/rem/rem0rec.cc

```diff
@@ -1850,12 +1843,12 @@ rec_copy_prefix_to_buf(
 	}
 
 	if (rec_is_instant(rec)) {
-		ut_ad(!dict_table_is_zip(index->table) && !dict_index_is_ibuf(index));
+		ut_ad(!dict_index_is_ibuf(index));
```

Comment:
I would suggest the following:
```
inline bool rec_is_instant(
#ifdef UNIV_DEBUG
	const dict_index_t* index,
#endif
	const rec_t* rec)
{
	…
}
#ifndef UNIV_DEBUG
# define rec_is_instant(rec, index) rec_is_instant(rec)
#endif
```
Then, inside the function, assert `ut_ad(index->is_instant())` and possibly `ut_ad(!dict_index_is_ibuf(index))` if the flag is set in the record.